### PR TITLE
5. Patch Requests

### PR DESCRIPTION
**Pull request by michaeljames24**
Ticket 5. Patch requests for reviews.

### CODE REVIEW

File: controllers/patch_controllers.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const {updateReview} = require('../models/patch_models');
+
+exports.patchReview = (req, res, next) => {
+    const { inc_votes } = req.body;
+    const { review_id } = req.params;
+    updateReview(inc_votes, review_id)
+    .then(review => {
+        res.status(200).send(review[0]);
```
Comment (CatHoang) [inline controllers/patch_controllers.js:8]: would we just want to send review[0] or what we want to send a object with a key of review?

**Review by CatHoang — COMMENTED** [review]
Hi Michael,

Some good code, just a couple of suggestions: 
In your controller, is review on a key of review? How are you sending it back at the moment?

Could we test for what happens if there are no votes on the sent object or if we send an empty object?

File: controllers/patch_controllers.js (hunk 1 above)
Comment (michaeljames24) [reply controllers/patch_controllers.js]: Thanks Cat. I've amended the patch request output, so it now returns the review object on a key of review. I've also added a test for invalid inputs on patch requests (specifically empty objects). Maybe when I'm further along and I know exactly what users will be allowed to patch, I could change it so the request only goes through if the input object contains one or more of specific properties? Cheers

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,28 +46,109 @@ describe("GET/api/categories requests.", () => {
...
+
+    test("Responds to empty object on patch request with 400 status and relevant message.", () => {
+        return request(app).patch('/api/reviews/2')
+        .expect(400)
+        .send({})
+        .then(({body}) => {
+            expect(body.message).toBe("Invalid input object.");
+        })
+    })
+
```
Comment (emilyb93) [inline __tests__/app.test.js:153]: What happens if we try to patch a url where the id is the incorrect data type?

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Id like you to restructure your test suite.

 Its quite hard to go back and forth between error sections and the happy path sections. 

Could you please keep the error handling for a path/endpoint to the same describe block. 

You can nest a describe inside of a describe, so why dont we try a describe block to hold all of the tests for one endpoint, and inside of it; a describe for the happy path tests, and a describe for the error handling for that path/method.

 It will make it much easier to read and review

File: __tests__/app.test.js (hunk 2 above)
Comment (michaeljames24) [reply __tests__/app.test.js:173]: Thanks Emily. I've restructured the test suites now. If the user uses an invalid review id for a patch request they should now get a 400 status and a 'Invalid Review ID' message.

**Comment by Dutton1990** [comment]
Much neater Michael! Looks good to me 👍
